File: src/operations/MultiplyOperation.js
import { Operation } from './Operation.js';

/**
 * Multiplication operation.
 * Follows Single Responsibility Principle - only handles multiplication.
 */
export class MultiplyOperation extends Operation {
  execute(a, b) {
    return a * b;
  }

  getSymbol() {
    return '*';
  }
}

File: src/operations/multiply_operation.py
"""
Multiplication operation.
Follows Single Responsibility Principle - only handles multiplication.
"""
from .operation import Operation


class MultiplyOperation(Operation):
    """Multiplication operation implementation."""

    def execute(self, a: float, b: float) -> float:
        """Performs multiplication of two operands."""
        return a * b

    def get_symbol(self) -> str:
        """Returns the multiplication symbol."""
        return '*'

